test/integration/scheduler_perf/scheduler_perf.go

```diff
@@ -756,8 +756,8 @@ func RunBenchmarkPerfScheduling(b *testing.B, outOfTreePluginRegistry frameworkr
 					// etcd.
 					framework.StartEtcd(b, output)
 
-					// 30 minutes should be plenty enough even for the 5000-node tests.
-					timeout := 30 * time.Minute
```

Comment:
I checked the scheduler_perf log, I think the root cause is we're timeout waiting for the pod to be scheduled, code here:
https://github.com/kubernetes/kubernetes/blob/9d8edcac2f3d2b573d0f32a20f2e51cb5268907d/test/integration/scheduler_perf/scheduler_perf.go#L1309

Also see the error log:
    `scheduler_perf.go:1017: FATAL ERROR: op 9: error in waiting for pods to get scheduled: at least pod test/test-drah9mnh is not scheduled: context deadline exceeded`

I do check the time, it tells the truth, and we stuck in this ` scheduler_perf.go:1323: namespace: test, pods: want 1000, got 996` for almost 10 mins, something wrong here, the last 4 pods can't be scheduled.

So I think extending the time doesn't work.

Reply:
Maybe https://github.com/kubernetes/kubernetes/pull/125360 can help with this.

Reply:
If pods get moved to the backoff queue, they can get stuck for a long time until the retry timeout happens.

I can't reproduce that locally and it worked before, so bumping up the timeout is just a way to check whether it just became a little bit slower. We can revert if it doesn't help.

Reply:
I'm still waiting for https://github.com/kubernetes/test-infra/pull/32707 to get merged. That would be a way to test in a pre-submit.

Reply:
/test pull-scheduler-perf

Reply:
@kerthcet: you are right, extending the timeout won't help.

I now have a log at -v5: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/125313/pull-kubernetes-scheduler-perf/1801955385091821568/build-log.txt

The failure there is:
```
    scheduler_perf.go:1017: FATAL ERROR: op 7: error in waiting for pods to get scheduled: at least pod test/test-dra8sj4s is not scheduled: context deadline exceeded
```

Searching for that pod and the associated ResourceClaim shows:
```console
$ grep -e test/test-dra8sj4s -e test/test-dra8sj4s-resource-zxm9r build-log.txt 
    graph_builder.go:692: I0615 12:34:14.013092] GraphBuilder process object apiVersion="v1" kind="Pod" object="test/test-dra8sj4s" uid="99b17925-47e6-4afc-9935-2f9685f7abc8" eventType="add" virtual=    <bool>: false
    eventhandlers.go:128: I0615 12:34:14.018764] Add event for unscheduled pod pod="test/test-dra8sj4s"
I0615 12:34:14.018781   27079 dynamicresources.go:474] "pod unschedulable" reason="pod \"test/test-dra8sj4s\": ResourceClaim not created yet"
I0615 12:34:14.018790   27079 scheduling_queue.go:539] "Status after running PreEnqueue plugin" pod="test/test-dra8sj4s" plugin="DynamicResources" status="pod \"test/test-dra8sj4s\": ResourceClaim not created yet"
    controller.go:557: I0615 12:34:19.163726] checking pod="test/test-dra8sj4s" podClaim="resource"
    controller.go:660: I0615 12:34:19.171506] Created ResourceClaim podClaim="resource" claim="test/test-dra8sj4s-resource-zxm9r" pod="test/test-dra8sj4s"
    graph_builder.go:692: I0615 12:34:19.180410] GraphBuilder process object apiVersion="v1" kind="Pod" object="test/test-dra8sj4s" uid="99b17925-47e6-4afc-9935-2f9685f7abc8" eventType="update" virtual=    <bool>: false
    eventhandlers.go:151: I0615 12:34:19.181285] Update event for unscheduled pod pod="test/test-dra8sj4s"
I0615 12:34:19.181378   27079 scheduling_queue.go:539] "Status after running PreEnqueue plugin" pod="test/test-dra8sj4s" plugin="DynamicResources" status="resourceclaim.resource.k8s.io \"test-dra8sj4s-resource-zxm9r\" not found"
    controller.go:509: I0615 12:34:19.184186] nothing to do for pod, scheduler will deal with it pod="test/test-dra8sj4s"
    graph_builder.go:692: I0615 12:34:19.188535] GraphBuilder process object apiVersion="resource.k8s.io/v1alpha2" kind="ResourceClaim" object="test/test-dra8sj4s-resource-zxm9r" uid="36380712-b9fb-4c38-a264-9e61e08d28e7" eventType="add" virtual=    <bool>: false
    controller.go:304: I0615 12:34:19.190391] test-driver.cdi.k8s.io/resource controller: Adding new work item key="claim:test/test-dra8sj4s-resource-zxm9r"
    controller.go:557: I0615 12:34:19.948900] checking pod="test/test-dra8sj4s" podClaim="resource"
    controller.go:588: I0615 12:34:19.948940] claim already created pod="test/test-dra8sj4s" podClaim="resource" resourceClaim="test-dra8sj4s-resource-zxm9r"
    controller.go:509: I0615 12:34:19.948966] nothing to do for pod, scheduler will deal with it pod="test/test-dra8sj4s"
    controller.go:841: I0615 12:34:19.949104] claim reserved for counts currentCount=    <int>: 0 claim="test/test-dra8sj4s-resource-zxm9r" updatedCount=    <int>: 0 builtinController=    <bool>: false
    controller.go:375: I0615 12:34:20.649171] test-driver.cdi.k8s.io/resource controller: processing key="claim:test/test-dra8sj4s-resource-zxm9r"
    controller.go:519: I0615 12:34:20.649198] test-driver.cdi.k8s.io/resource controller: ResourceClaim waiting for first consumer key="claim:test/test-dra8sj4s-resource-zxm9r"
    controller.go:379: I0615 12:34:20.649222] test-driver.cdi.k8s.io/resource controller: completed key="claim:test/test-dra8sj4s-resource-zxm9r"
    scheduler_perf.go:1017: FATAL ERROR: op 7: error in waiting for pods to get scheduled: at least pod test/test-dra8sj4s is not scheduled: context deadline exceeded
```

The relevant part is this:
```
    eventhandlers.go:151: I0615 12:34:19.181285] Update event for unscheduled pod pod="test/test-dra8sj4s"
I0615 12:34:19.181378   27079 scheduling_queue.go:539] "Status after running PreEnqueue plugin" pod="test/test-dra8sj4s" plugin="DynamicResources" status="resourceclaim.resource.k8s.io \"test-dra8sj4s-resource-zxm9r\" not found"
```

This could be the race between event handler triggered by informer and the plugin using the assume cache. The fix is in https://github.com/kubernetes/kubernetes/pull/124595. Let's see whether the new job passed there...